### DISCUSSION TITLE
VCC-GND pyboard F407VGT6 (old blue colored version)

### DISCUSSION THREAD

**Discussion by johnsfchan** [micropython/micropython#14359] [STM32 / Pyboard]
I am a newbie and have purchased subject board recently.

When I first plug the board to my Windows 11 PC, it ran fine with 'MicroPython v1.11 on 2019-09-20; YD-407VG-PYBv1.1 vcc-gnd.com with STM32F407VG'.

However, it failed to start after I had upgraded the firmware to latest 'VCC_GND_F407VE-20240222-v1.22.2' using STM32CubeProgrammer.  The symptoms are (1) not shown as one of Windows COMxx (2) an unknown USB device detected with no valid driver.

I had contacted VCC-GND.com but the technical support team told me that firmware version MicroPython v1.11 is too old and cannot be found anymore from their end.

Please kindly advise if I can download that old version of MicroPython from micropython.org .

Thanks in advance.

**Comment by peterhinch** [comment]
If VCC-GND are supplying badly pirated Pyboards which can't run the latest firmware, surely it is up to them to supply a build of firmware that does work.

The [real Pyboard](https://store.micropython.org/product/PYBv1.1) will run the latest firmware.

**Comment by projectgus** [comment]
MicroPython has added support for some VCC-GND products, but they're all the STM32 "mini" boards that are simple breakouts for the microcontroller. This is probably why the official build "'VCC_GND_F407VE-20240222-v1.22.2" doesn't work, it's for a different product (also the -VE has less flash than -VG, but I think that's not the issue here.)

The build string on the working board you have is "YD-407VG-PYBv1.1 vcc-gnd.com with STM32F407VG". This isn't from any official MicroPython source version, which suggests this version is something vcc-gnd created and built themselves. As @peterhinch says, you'll need to ask them where to find a new firmware that works with it (or switch to a genuine pyBoard).
